pkg/apis/walruscore/v1/resource_run.go

```diff
@@ -41,3 +108,174 @@ type ResourceRunList struct {
 }
 
 var _ runtime.Object = (*ResourceRunList)(nil)
+
+// ResourceRunStep stores the step run result of the resource run.
+type ResourceRunStep struct {
+	// Name is the name of the step.
+	Name string `json:"name"`
+
+	// Type is the type of the step.
+	Type string `json:"type"`
```

Comment:
Is this ResourceRunJobType?
We have RunJobType, RunTaskType, RunStep, are they refering to the same concept?
How does the `Type` work with `ResourceRunStepTemplate `?

Reply:
Let's clear the concepts here.

RunJobType --> RunOperationType (plan, apply).  It is walrus defined system resource operation.

RunStepType -> plan, apply, hook step types. `RunStepType` refers to the step types—namely plan, apply, and hooks. Its purpose is to record the results from system operations as well as from the hooks.

In the case of `ResourceRunStepTemplate`, we do not have a distinct `type` field. Instead, we differentiate between RunStepTypes by utilizing the specification (spec) outlined, when Container field is not empty, the step is `container` type step.